### PR TITLE
Issue113 - Backend accepts commit message as an optional parameter

### PR DESCRIPTION
**Pull request by ilkkamaksy**
Closes issue #113

### CODE REVIEW

File: backend/src/utils/sanitize.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+  
+export const sanitizeBranchName = (inputStr: string):string => {
+  // Rules for branch names: https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
+  let str = inputStr.trim()
+  str = stripAllSpecialCharsButDash(str)
+  str = stripConsecutiveSpecialChars(str)
+  str = stripStartingDashFromString(str)
+  return str
+}
```
Comment (guotin) [inline backend/src/utils/sanitize.ts:15]: Would use something cleaner here at some point. Maybe some `reduce` magic could be used in the future.

**Review by guotin — APPROVED** [review]
The sanitizing of commit messages and branch name could confuse the user. Do you have any plans for this? But we can go with this for now imo.

File: backend/src/utils/sanitize.ts (hunk 1 above)
Comment (tulma95) [reply backend/src/utils/sanitize.ts:15]: I don't like using the let. I'd recommend we go more into functional programming and use `pipe` [(More info)](https://www.freecodecamp.org/news/pipe-and-compose-in-javascript-5b04004ac937/). We could write our own generic pipe function or use  functional javascript library like ramda or lodash

**Comment by tulma95** [comment]
Sanitize functions should be easy to test with jest. I think we should add the tests too to this pull request

**Review by att78 — APPROVED** [review]

**Comment by guotin** [comment]
Wait for @tulma95's change for sanitize funcs

**Pull request merged**